### DISCUSSION TITLE
Fire an event once migration command complete

### DISCUSSION THREAD

**Discussion by mdariftiens** [laravel/framework#51156] [Q&A] [answered]
I need to fire an event after completing `php artisan migrate` . Suppose, I need to run a command or sending email after completing migration. Is there any way to do this? If exists, then how can I achieve my purpose. if not exists, can we add a config file to hold an event then fire event after complete command?

**Comment by pintend** [comment]
https://laravel.com/docs/11.x/migrations#events

**Reply by mdariftiens** [reply]
But how can I code in `Illuminate\Database\Events\MigrationsEnded` . directly code in `Illuminate\Database\Events\MigrationsEnded` is not good idea. now how can i implement my logic ? can help please?

**Reply by rdgout** [reply, marked as answer]
Your question was if there is an event and there is. The next step is for you to implement using the event.
Since you don't seem to fully grasp the idea of events read through the documentation [here](https://laravel.com/docs/11.x/events#introduction)

**Reply by mdariftiens** [reply]
I got the idea now. Thanks @pintend @rdgout .
